cluster/kube-up.sh

```diff
@@ -32,6 +32,31 @@ fi
 
 source "${KUBE_ROOT}/cluster/kube-util.sh"
 
+DEPRECATED_PROVIDERS=(
+  "centos"
+  "libvert-coreos"
+  "local"
+  "openstack-heat"
+  "photon-controller"
+  "vagrant"
+  "vsphere"
+  "windows"
+)
+
+for provider in "${DEPRECATED_PROVIDERS[@]}"; do
+  if [[ "${KUBERNETES_PROVIDER}" == "${provider}" ]]; then
+    cat <<EOF 1>&2
+
+!!! DEPRECATION NOTICE !!!
+
+The '${provider}' kube-up provider is deprecated and will be removed in a future
```

Comment:
Is there an issue or announcement we can link to here to provide more context about the deprecation? Also, if someone sees this message and immediately panics and wants to convince us **not** to remove this support, how should they proceed? A little clarity up front will help us organize any support pings we get in the future.